ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -12975,8 +12982,16 @@ void HqlCppTranslator::doBuildAggregateProcessTransform(BuildCtx & ctx, BoundRow
             }
             break;
         }
+
+        if (isAggregateOp && isDynamicOffset)
+            throwError1(HQLERR_AggregateDyanmicOffset, str(target->queryChild(1)->queryId()));
+
         if (target->queryType()->getSize() == UNKNOWN_LENGTH)
+        {
            isVariableOffset = true;
+           if (isAggregateOp)
+               isDynamicOffset = true;
```

Comment:
Good question.  I'll think again - there may still be some cases that have problems.  It should be reporting an error if any expression follows a variable size aggregate.  reorderAggregateFields() should probably catch a couple more cases.